website/docs/gator.md

```diff
@@ -0,0 +1,180 @@
+---
+id: gator
+title: The gator CLI
+---
+
+The `gator` CLI is a tool for testing Gatekeeper ConstraintTemplates and
+Constraints on Kubernetes objects.
+
+We plan on adding more subcommands in the future.
+
+## Installation
+
+To install `gator`, you may either
+[download the binary](https://github.com/open-policy-agent/gatekeeper/releases/tag/v3.7.0)
+relevant to your system or build it directly from source.
+
+To build from source:
+```
+go get github.com/open-policy-agent/gatekeeper/cmd/gator
+```
+
+## The `gator test` subcommand
+
+### Writing Test Suites
+
+`gator test` organizes tests into three levels: Suites, Tests, and Cases:
+
+- A Suite is a file which defines Tests.
+- A Test declares a ConstraintTemplate, a Constraint, and Cases to test the
+  Constraint.
+- A Case defines an object to validate and whether the object is expected to
+  pass validation.
+
+Any file paths declared in a Suite are assumed to be relative to the Suite
+itself. Absolute paths are not allowed. Thus, it is possible to move around a
+directory containing a Suite, and the files it uses for tests.
+
+### Suites
+
+[An example Suite file](https://github.com/open-policy-agent/gatekeeper-library/blob/8765ec11c12a523688ed77485c7a458df84266d6/library/general/allowedrepos/suite.yaml).
+
+To be valid, a Suite file must declare:
+```yaml
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+```
+
+This differentiates Suites from "real" Kubernetes objects. `gator test` silently
+ignores files which do not declare these and does not attempt to run them.
+
+A Suite may declare multiple Tests, each containing different Templates and
+Constraints. Each Test in a Suite is independent.
+
+### Tests
+
+The `tests` top-level field is a list representing tests to run.
+
+A Test compiles a ConstraintTemplate, and instantiates a Constraint for the
+ConstraintTemplate. It is an error for the Constraint to have a different type
+than the ConstraintTemplate, or for the ConstraintTemplate to not compile.
+
+### Cases
+
+The `cases` field is within `tests`, is a list.
+
+A Case validates an object against a Constraint. The case may specify that the
+object is expected to pass or fail validation, and may make assertions about
+the returned violations (if any).
+
+A Case must specify `assertions` and whether it expects violations. The simplest
+way to declare this is:
+
+The Case expects at least one violation:
+```yaml
+assertions:
+- violations: yes
+```
+
+The Case expects no violations:
+```yaml
+assertions:
+- violations: no
+```
+
+- `violations` is either "yes", "no", or a non-negative integer. If "yes", at
+  least one violation must otherwise match the assertion. If "no", then no
+  violation messages must otherwise match the assertion. If a nonnegative
+  integer, then exactly that many violations must match.
+- `message` matches violations containing the exact string specified. `message`
```

Comment:
maybe add something about what it means to provide no message?

Reply:
Done!